### PR TITLE
Add 'recovering' status

### LINKED ISSUE #1779: [Recovery] Add `recovering` state
Add the state to the proto enum [`Status`](https://github.com/ConduitIO/conduit/blob/d4e617775fb05e20261389da31a7506566e8b47b/proto/api/v1/api.proto#L14-L22) as well as to the [converter](https://github.com/ConduitIO/conduit/blob/d4e617775fb05e20261389da31a7506566e8b47b/pkg/web/api/toproto/pipeline.go#L45).

As part of this issue we should also allow the state in the [`Stop`](https://github.com/ConduitIO/conduit/blob/d4e617775fb05e20261389da31a7506566e8b47b/pkg/pipeline/lifecycle.go#L127) function (a recovering pipeline should be able to stop), also check if there are other places where we need to include it.

### PR DESCRIPTION
**Pull request by hariso**
### Description

Closes #1779.

### Quick checks

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: pkg/pipeline/lifecycle.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ func (s *Service) Stop(ctx context.Context, pipelineID string, force bool) error
 		return err
 	}
 
-	if pl.Status != StatusRunning {
+	if pl.Status != StatusRunning && pl.Status != StatusRecovering {
```
Comment (hariso) [inline pkg/pipeline/lifecycle.go:127]: It should be possible to stop a recovering pipeline.

File: Makefile [hunk 2]
```diff
@@ -51,7 +51,7 @@ run:
 
 .PHONY: proto-generate
 proto-generate:
-	rm -rf proto/gen && cd proto && buf generate
```
Comment (raulb) [inline Makefile:54]: Good catch

File: pkg/pipeline/lifecycle.go [hunk 3]
```diff
@@ -138,7 +138,10 @@ func (s *Service) Stop(ctx context.Context, pipelineID string, force bool) error
 }
 
 func (s *Service) stopGraceful(ctx context.Context, pl *Instance, reason error) error {
-	s.logger.Info(ctx).Str(log.PipelineIDField, pl.ID).Msg("gracefully stopping pipeline")
+	s.logger.Info(ctx).
```
Comment (raulb) [inline pkg/pipeline/lifecycle.go:141]: I wonder if we need a `Stopping` pipeline status.

File: .golangci.yml [hunk 4]
```diff
@@ -71,7 +71,7 @@ linters:
     - errcheck
     - errname
     # - errorlint
-    # - exhaustive
+    - exhaustive
```
Comment (raulb) [inline .golangci.yml:74]: @hariso What made you realize we needed to enable this back?

Comment (hariso) [reply .golangci.yml:74]: In one part of the code, there was a switch statement that was transforming proto statuses into Go statuses.

**Review by raulb — APPROVED** [review]

**Pull request merged**